### PR TITLE
Add example with-hls-js

### PR DESCRIPTION
**Pull request by dylanjha**
Follow-up from discussion https://github.com/zeit/next.js/pull/12289

This example uses:

  * [hls.js](https://github.com/video-dev/hls.js/)
  * For content -- the canonical video example from the video community "Big Buck Bunny" - hosted by Mux (https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8)

with-hls-js_2020-05-11_08-25-12

### CODE REVIEW

**Comment by dylanjha** [comment]
Thanks for taking a look @lfades! Let me know if there are any blockers to getting this in examples/!

File: examples/with-hls-js/.gitignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# debug
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
```
Comment (lfades) [inline examples/with-hls-js/.gitignore:30]:
```suggestion
.env*.local
```

File: examples/with-hls-js/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "with-hls-js",
+  "version": "0.1.0",
+  "private": true,
```
Comment (lfades) [inline examples/with-hls-js/package.json:4]:
```suggestion
  "license": "MIT",
```

File: examples/with-hls-js/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "hls.js": "^0.13.2",
+    "next": "9.3.6",
```
Comment (lfades) [inline examples/with-hls-js/package.json:12]:
```suggestion
    "next": "latest",
```

File: examples/with-hls-js/pages/api/hello.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+export default (req, res) => {
+  res.statusCode = 200
+  res.json({ name: 'John Doe' })
+}
```
Comment (lfades) [inline examples/with-hls-js/pages/api/hello.js:6]: Remove `pages/api` entirely, the example doesn't need API routes.

**Review by lfades — CHANGES_REQUESTED** [review]
@dylanjha Thank you for the PR, please check my suggestion and then I'll merge.

**Comment by dylanjha** [comment]
@lfades thanks! Updated.. That one "azure next.js" check seems to be a little flakey

**Review by lfades — APPROVED** [review]
Niceeee, thank you!

**Comment by jcohenho** [comment]
I'm trying to follow this example but I keep getting the error `Uncaught ReferenceError: window is not defined`. I was under the impression that using hooks like `useEffect` would give us access to window, and in fact I do have access to window, yet I still see this error message. Where in the chain would window be undefined by hls.js?

**Pull request merged**